### DISCUSSION TITLE
Spacing of nodes in bipartite graph

### DISCUSSION THREAD

**Discussion by sojohan** [networkx/networkx#6241] [Q&A]
Hi Experts 
I have this code snippet to make bipartite graph. 
`G = nx.Graph( node_size=1,node_label_offset=10)
G.add_nodes_from(freqsel.prod_name,size=10, bipartite=0)
G.add_nodes_from(freqsel.pattern, bipartite=1)
#G.add_edges_from(freq2items40above.support)
#G.add_edges_from(support)
G.add_edges_from(edges)
pos = nx.bipartite_layout(G, freqsel.prod_name,scale=0.01,aspect_ratio=10)
nx.draw(G, width=support,node_size=1,edge_color=support,pos=pos,edge_cmap=cmap,with_labels=True, font_size=13)
sm = plt.cm.ScalarMappable(cmap=cmap, norm=plt.Normalize(vmin = vmin, vmax=vmax))
sm._A = []
plt.colorbar(sm)

plt.title('40 Above',fontsize=20)
plt.show()

`
My questions is if I can increase the spacing on right side. See the picture below.
And the result is : 
<img width="583" alt="image" src="https://user-images.githubusercontent.com/13655240/204307218-f706fe8c-8aab-46a1-92d2-01a8f85f1617.png">

**Comment by rossbar** [comment]
There's a few different things that might help. Note that the scales for everything are relative, so doing something like reducing the font_size will help with the labels running over each other, e.g.

```python
>>> nx.draw_networkx_labels(G, pos, font_size=10)
```

Note also that the return value of the layout functions is a dict keyed by node where the values are 2D arrays representing the coordinates of the nodes. Therefore you can customize the node positions however you like. For example, if you want to increase the vertical scale of the larger node set, you could do something like:

```python
>>> for node in partition_with_more_nodes:
...     pos[n][-1] *= 5
```

Some combination of these two tweaks should get you closer to your desired output.
